crates/loader/src/local.rs

```diff
@@ -374,13 +374,29 @@ impl LocalLoader {
                     quoted_path(&dest_parent)
                 )
             })?;
-        crate::fs::copy(src, dest).await.with_context(|| {
-            format!(
-                "Failed to copy {} to {}",
-                quoted_path(src),
-                quoted_path(dest)
-            )
-        })?;
+
+        let failed_to_copy_error = format!(
+            "Failed to copy {} to {}",
+            quoted_path(src),
+            quoted_path(dest)
+        );
+
+        let dest_str = dest
+            .to_str()
+            .expect("Could not convert destination path to str");
+
+        if dest_str.ends_with(std::path::MAIN_SEPARATOR)
+            || dest.to_str().unwrap_or_default().ends_with('.')
+            || dest.to_str().unwrap_or_default().ends_with("..")
+        {
+            bail!(format!(
+                "{failed_to_copy_error}: Illegal destination filename"
```

Comment:
If I'm reading this right, `failed_to_copy_error` will have the _host_ path, so the user will be getting told that `/tmp/gibber/gibber/gibber/` is illegal, which they will go "true but I never asked for that path, what gives."  What we want them to understand is that the _specified destination_ in the `spin.toml` is illegal.

(I may be misunderstanding though, feel free to tick me off)